ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PushQueryPublisher.java

```diff
@@ -53,67 +53,33 @@
 
   private static final Logger log = LoggerFactory.getLogger(PushQueryPublisher.class);
 
-  private final KsqlEngine ksqlEngine;
-  private final ServiceContext serviceContext;
-  private final ConfiguredStatement<Query> query;
-  private final Optional<LocalCommands> localCommands;
   private final ListeningScheduledExecutorService exec;
-  private final PushRouting pushRouting;
-  private final boolean isScalablePush;
-  private final Context context;
-  private final SlidingWindowRateLimiter scalablePushBandRateLimiter;
+  private final PushQueryMetadata queryMetadata;
 
   private PushQueryPublisher(
-      final KsqlEngine ksqlEngine,
-      final ServiceContext serviceContext,
       final ListeningScheduledExecutorService exec,
-      final ConfiguredStatement<Query> query,
-      final Optional<LocalCommands> localCommands
-  ) {
-    this.ksqlEngine = requireNonNull(ksqlEngine, "ksqlEngine");
-    this.serviceContext = requireNonNull(serviceContext, "serviceContext");
-    this.exec = requireNonNull(exec, "exec");
-    this.query = requireNonNull(query, "query");
-    this.localCommands = requireNonNull(localCommands, "localCommands");
-    this.pushRouting = null;
-    this.isScalablePush = false;
-    this.context = null;
-    this.scalablePushBandRateLimiter = null;
-  }
-
-  private PushQueryPublisher(
-      final KsqlEngine ksqlEngine,
-      final ServiceContext serviceContext,
-      final ListeningScheduledExecutorService exec,
-      final ConfiguredStatement<Query> query,
-      final ImmutableAnalysis analysis,
-      final PushRouting pushRouting,
-      final Context context,
-      final SlidingWindowRateLimiter scalablePushBandRateLimiter
+      final PushQueryMetadata queryMetadata
   ) {
-    this.ksqlEngine = requireNonNull(ksqlEngine, "ksqlEngine");
-    this.serviceContext = requireNonNull(serviceContext, "serviceContext");
-    this.exec = requireNonNull(exec, "exec");
-    this.query = requireNonNull(query, "query");
-    this.localCommands = Optional.empty();
-    this.pushRouting = requireNonNull(pushRouting, "pushRouting");
-    this.isScalablePush = true;
-    this.context = requireNonNull(context, "context");
-    this.scalablePushBandRateLimiter =
-        requireNonNull(scalablePushBandRateLimiter, "scalablePushBandRateLimiter");
+    this.exec = exec;
+    this.queryMetadata = queryMetadata;
   }
 
-  public static PushQueryPublisher createPublisher(
+  public static PushQueryPublisher createPushQueryPublisher(
       final KsqlEngine ksqlEngine,
       final ServiceContext serviceContext,
       final ListeningScheduledExecutorService exec,
       final ConfiguredStatement<Query> query,
       final Optional<LocalCommands> localCommands
```

Comment:
High level description of this refactor:

We were previously capturing all these arguments as instance fields, and an overlapping set of arguments for the scalable push query publisher before, and then branching later on in `subscribe` to use those fields to create a PushQueryMetadata of the appropriate type.

But after that branch, the only thing we needed was the query metadata and the executor service. So this PR cuts out the middle man: we just go ahead and create that metadata up front in the static factor methods so that we neither need that big list of fields nor the branched logic later on.